tpr/tpr.go

```diff
@@ -0,0 +1,120 @@
+package tpr
+
+import (
+	"fmt"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"github.com/rossf7/pingdom-operator/pkg/util"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	tprInitRetries    = 30
+	tprInitRetryDelay = 3 * time.Second
+)
+
+type Config struct {
+	// Dependencies.
+	Clientset kubernetes.Interface
+	Rest      rest.Interface
+
+	// Settings.
+	Name        string
+	Group       string
+	Version     string
+	Description string
+}
+
+type tpr struct {
+	clientset kubernetes.Interface
+
+	name          string
+	group         string
+	version       string
+	description   string
+	qualifiedName string
+
+	endpointList string
+}
+
+func New(config Config) (*tpr, error) {
```

Comment:
Why should we return a private type? The default godoc will not show anything in case the constructor returns a private type, even if the methods on it are public. I would say the type should be public and properties not allowed to be exposed should just be private.